server/controllers/documents/find.js

```diff
@@ -0,0 +1,151 @@
+import Documents from '../../models/documents.js';
+import authorised from '../helpers/authorise.js';
+
+const find = {};
+
+find.id = (req, res) => {
+  Documents.findById(req.params.id, (error, document) => {
+    // Inform user of errors with the database
+    if (error) {
+      return res.status(500).json({
+        success: false,
+        message: 'There was an error reading from the database'
+      });
+    }
+
+    // Document returieved, authorise before sending to user
+    if (document) {
+      if (authorised(req, document)) {
+        return res.json(document);
+      }
+      // User not authorized
+      return res.status(401).json({
+        success: false,
+        message: 'Not authorized to access document'
+      });
+    }
+    // Failed, no document with specified ID
+    return res.status(404).json({
+      success: false,
+      message: 'Document does not exist',
+    });
+  });
+};
+
+// Retrieve all documents
+find.all = (req, res) => {
```

Comment:
*Cod quality - Nitpick*

Function is too long, have a helper function that uses a switch case.